osu.Game/Screens/Select/FooterButton.cs

```diff
@@ -133,5 +133,17 @@ protected override bool OnKeyDown(KeyDownEvent e)
 
             return base.OnKeyDown(e);
         }
+
+        public void FadeText(string newText, double duration = 0d, Easing easing = Easing.None)
+        {
+            spriteText?.FadeOut(duration, easing).Expire();
+            Add(spriteText = new OsuSpriteText
```

Comment:
Wouldn't this create a new `SpriteText` every time you press or release the key? Given that this now seems to change even when not visible it could create quite a few of them in SongSelect. Maybe just have the states and show the  correct one depending on the state?

Reply:
The old one is removed from the scene graph once it faded out completely. I'll rework this however.